src/Plugin.php

```diff
@@ -24,7 +24,7 @@
      * @param callable         $next    Next middleware in the chain, the request is passed as the first argument
      * @param callable         $first   First middleware in the chain, used to to restart a request
      *
-     * @return Promise
+     * @return Promise Resolves a PSR-7 Response or fails with an Http\Client\Exception (The same as HttpAsyncClient).
```

Comment:
Add "@see http://docs.php-http.org/en/latest/plugins/build-your-own.html". I've added some lines about exceptions (https://github.com/php-http/documentation/pull/151)

> Is this a BC break ?

No, this is not a BC break.